python/ml4ir/applications/ranking/model/metrics/metrics_impl.py

```diff
@@ -297,3 +302,257 @@ def _get_matches_hook(self, y_pred_click_ranks):
             Ranks tensor cast to float
         """
         return tf.cast(y_pred_click_ranks, tf.float32)
+
+
+class RankMatchFailure(MeanMetricWrapper, CombinationMetric):
+    def __init__(
+        self,
+        feature_config: FeatureConfig,
+        metadata_features: Dict,
+        state: str = MetricState.NEW,
+        name="RankMatchFailure",
+        dtype: Optional[dtypes.DType] = None,
+        **kwargs,
+    ):
+        """
+        Creates a `RankMatchFailure` instance to compute mean of rank
+
+        Parameters
+        ----------
+        name : str
+            string name of the metric instance.
+        dtype : str, optional
+            data type of the metric result.
+        rank : Tensor object
+            2D tensor representing ranks/rankitions of records in a query
+        mask : Tensor object
+            2D tensor representing 0/1 mask for padded records
+
+        Notes
+        -----
+        rank and mask should be same shape as y_pred and y_true
+
+        This metric creates two local variables, `total` and `count` that are used to
+        compute the frequency with which `y_pred` matches `y_true`. This frequency is
+        ultimately returned as `categorical accuracy`: an idempotent operation that
+        simply divides `total` by `count`.
+        `y_pred` and `y_true` should be passed in as vectors of probabilities, rather
+        than as labels. If necessary, use `tf.one_hot` to expand `y_true` as a vector.
+        If `sample_weight` is `None`, weights default to 1.
+        Use `sample_weight` of 0 to mask values.
+        """
+        name = "{}_{}".format(state, name)
+        # TODO: Handle Example dataset without mask and rank fields
+        rank = metadata_features[feature_config.get_rank("node_name")]
+        mask = metadata_features[feature_config.get_mask("node_name")]
+        if not feature_config.aux_label:
```

Comment:
Dependency on the aux_label can be relaxed to allow any field.

Limitation currently is the callbacks and the feature config population